pallets/api/src/messaging/mod.rs

```diff
@@ -615,6 +615,15 @@ pub mod pallet {
 			xcm_response: Response,
 		) -> DispatchResult {
 			T::XcmResponseOrigin::ensure_origin(origin)?;
+
+			// Manually adjust weight ahead of fallible execution.
+			// The fees of which should have been paid.
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
```

Comment:
Similar to how you did it somewhere else, shouldn't we check if the weight is already exceeded before doing this unchecked increase?

Also, it would be good to add a comment why this is charged in addition to the xcm_response in `#[pallet::weight({`

Reply:
Yes great catch!